### DISCUSSION TITLE
S3 Mini ?

### DISCUSSION THREAD

**Discussion by SunnyMoo** [awawa-dev/HyperSerialESP32#46] [Q&A] [answered]
Hi,
can't find LOLIN S2 Mini anymore, seems it has been replaced by LOLIN S3 Mini.
Can we use S2 firmwares for S3 or we have to use generic ESP32 ones instead ?
Thanks and regards.

**Comment by awawa-dev** [comment, marked as answer]
Hi
No, it won't work. In the case of expressif, the issue with the model is quite complicated, because a higher model does not mean that it offers identical or better capabilities, as it is usually the other way around (at least when it comes to LED support). So S2 offers less than the original ESP32 but still provides some backward compatibility, and S3 turns it all upside down. My advice: if you can't find a recommended esp32/esp32-s2, stay away from them and choose a model with rp2040: preferably with a built-in level shifter, but basically any generic, non-custom version will be compatible.

**Comment by SunnyMoo** [comment]
Thanks for your lightning fast reply ! 👍
Best regards.
